autosubmit/helpers/autosubmit_helper.py

```diff
@@ -84,29 +83,25 @@ def handle_start_after(start_after, expid, BasicConfig):
             Log.critical(
-                "Experiment {0} is running a database version which is not supported by the completion trigger function. An updated DB version is needed.".format(
+                "Experiment {0} is running a database version which is not supported by the completion trigger "
+                "function. An updated DB version is needed.".format(
                     start_after))
-            return
-        Log.info(
-            "Autosubmit will start monitoring experiment {0}. When the number of completed jobs plus suspended jobs becomes equal to the total number of jobs of experiment {0}, experiment {1} will start. Querying every 60 seconds. Status format Completed/Queuing/Running/Suspended/Failed.".format(
-                start_after, expid))
-        while True:
-            # Query current run
-            current_run = exp_history.manager.get_experiment_run_dc_with_max_id()
-            if current_run and current_run.finish > 0 and current_run.total > 0 and current_run.completed + current_run.suspended == current_run.total:
+            return None
+        Log.info("Autosubmit will start monitoring experiment {0}. When the number of completed jobs plus suspended"
+                 " jobs becomes equal to the total number of jobs of experiment {0}, experiment {1} will start. "
+                 "Querying every 60 seconds. Status format Completed/Queuing/Running/Suspended/Failed.".format(
+            start_after, expid))
+        while current_run := exp_history.manager.get_experiment_run_dc_with_max_id():
+            if (current_run and current_run.finish > 0 and current_run.total > 0
+                    and current_run.total == current_run.completed + current_run.suspended):
                 break
-            else:
-                sys.stdout.write(
-                    "\rExperiment {0} ({1} total jobs) status {2}/{3}/{4}/{5}/{6}".format(start_after,
-                                                                                          current_run.total,
-                                                                                          current_run.completed,
-                                                                                          current_run.queuing,
-                                                                                          current_run.running,
-                                                                                          current_run.suspended,
-                                                                                          current_run.failed))
-                sys.stdout.flush()
+            sys.stdout.write(
+                f"\rExperiment {start_after} ({current_run.total} total jobs) status {current_run.completed}/"
+                f"{current_run.queuing}/{current_run.running}/{current_run.suspended}/{current_run.failed}")
+            sys.stdout.flush()
             # Update every 60 seconds
             sleep(60)
+    return None
```

Comment:
Not needed. A function that returns `None` can be just left without a return statement.

Reply:
And can you check this again? This is not needed, and at least my IDE is not complaining. There is, however, in `test_autosubmit_helper.py` an assert checking that it's `None`. That assertion needs to be removed (a function that returns only `None` will return `None` forever, no need to check if in that test).